certora/helpers/MerkleTrees.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: GNU AGPLv3
+pragma solidity ^0.8.0;
+
+import "./MerkleTreeLib.sol";
+
+contract MerkleTrees {
+    using MerkleTreeLib for MerkleTreeLib.Node;
+    using MerkleTreeLib for MerkleTreeLib.Tree;
+
+    mapping(address => MerkleTreeLib.Tree) trees;
+
+    function newLeaf(
+        address treeAddress,
+        address addr,
+        uint256 value
+    ) public {
+        trees[treeAddress].newLeaf(addr, value);
+    }
+
+    function newInternalNode(
+        address treeAddress,
+        address parent,
+        address left,
+        address right
+    ) public {
+        trees[treeAddress].newInternalNode(parent, left, right);
+    }
+
+    function setRoot(address treeAddress, address addr) public {
+        trees[treeAddress].setRoot(addr);
+    }
+
+    function getRoot(address treeAddress) public view returns (address) {
+        return trees[treeAddress].getRoot();
+    }
+
+    function getLeft(address treeAddress, address addr) public view returns (address) {
+        return trees[treeAddress].getLeft(addr);
+    }
+
+    function getRight(address treeAddress, address addr) public view returns (address) {
+        return trees[treeAddress].getRight(addr);
+    }
+
+    function getValue(address treeAddress, address addr) public view returns (uint256) {
+        return trees[treeAddress].getValue(addr);
+    }
+
+    function getHash(address treeAddress, address addr) public view returns (bytes32) {
+        return trees[treeAddress].getHash(addr);
+    }
+
+    function isEmpty(address treeAddress, address addr) public view returns (bool) {
+        return trees[treeAddress].nodes[addr].isEmpty();
+    }
+
+    function isWellFormed(address treeAddress, address addr) public view returns (bool) {
+        return trees[treeAddress].isWellFormed(addr);
+    }
+
+    // Check that the nodes are well formed on the path from the root.
+    function wellFormedPath(
+        address treeAddress,
+        address addr,
+        bytes32[] memory proof
+    ) public view {
+        MerkleTreeLib.Tree storage tree = trees[treeAddress];
+
+        for (uint256 i = proof.length; ; ) {
+            require(tree.isWellFormed(addr));
```

Comment:
I don't think so, or it would not verify because we need this property for the root. This function starts by the root and then goes down the tree, so the root is checked first